connectors/sources/onedrive.py

```diff
@@ -113,34 +166,100 @@ def set_logger(self, logger_):
+        skipped_exceptions=NotFound,
     )
-    async def get(self, url_name):
-        url = parse.urljoin(BASE_URL, url_name)
+    async def get(self, url):
         access_token = await self.token.get()
         headers = {"authorization": f"Bearer {access_token}"}
-        async with self._get_session.get(url=url, headers=headers) as response:
-            yield response
+        try:
+            async with self.session.get(url=url, headers=headers) as response:
+                yield response
+        except ServerConnectionError:
+            await self.close_session()
+        except ClientResponseError as e:
+            if e.status == 429 or e.status == 503:
+                response_headers = e.headers or {}
+                retry_seconds = DEFAULT_RETRY_SECONDS
+                if "Retry-After" in response_headers:
+                    try:
+                        retry_seconds = int(response_headers["Retry-After"])
+                    except (TypeError, ValueError) as exception:
+                        self._logger.error(
+                            f"Error while reading value of retry-after header {exception}. Using default retry time: {DEFAULT_RETRY_SECONDS} seconds"
+                        )
+                else:
+                    self._logger.warning(
+                        f"Rate Limited but Retry-After header is not found, using default retry time: {DEFAULT_RETRY_SECONDS} seconds"
+                    )
+                self._logger.debug(
+                    f"Rate Limit reached: retry in {retry_seconds} seconds"
+                )
+
+                await self._sleeps.sleep(retry_seconds)
+                raise ThrottledError from e
+            elif e.status == 404:
+                raise NotFound from e
+            elif e.status == 500:
+                raise InternalServerError from e
+            else:
+                raise
+
+    async def paginated_api_call(self, url_name, **url_kwargs):
+        url = parse.urljoin(BASE_URL, ENDPOINTS[url_name].format(**url_kwargs))
+        url = f"{url}?$top={FETCH_SIZE}"
+
+        while True:
+            try:
+                async for response in self.get(
+                    url=url,
+                ):
+                    response_json = await response.json()
+                    result = response_json["value"]
+
+                    if len(result) == 0:
+                        return
+
+                    yield result
+
+                    url = response_json.get("@odata.nextLink")
+                    if not url:
+                        return
+            except Exception as exception:
+                self._logger.warning(
+                    f"Skipping data for type {url_name} from {url}. Exception: {exception}."
+                )
+                break
+
+    async def get_user_ids(self):
+        async for response in self.paginated_api_call(url_name=USERS):
+            for user_detail in response:
+                yield user_detail["id"]
```

Comment:
Sorry, I didn't make it explicit. I'd like to make it more general to return user instead of user id:

 ```suggestion
    async def list_users(self):
        async for response in self.paginated_api_call(url_name=USERS):
            for user_detail in response:
                yield user_detail
```